pkg/guardian/routeratelimit.go

```diff
@@ -1,5 +1,62 @@
 package guardian
 
+import (
+	"github.com/sirupsen/logrus"
+	"net/url"
+	"time"
+)
+
+// This file contains the implementation of a Route Rate Limiter.
+// This is achieved by writing custom implementations of the KeyFunc and LimitProvider that the GenericRateLimiter
+// struct depends upon.
+
+// RouteLimitProvider implements the LimitProvider interface.
+type RouteLimitProvider struct {
+	*RedisConfStore
+}
+
+// GetLimit gets the limit for a particular request's path.
+func (rlp *RouteLimitProvider) GetLimit(req Request) Limit {
+	reqUrl, err := url.Parse(req.Path)
+	if err != nil || reqUrl == nil {
+		rlp.logger.Warnf("unable to parse url from request: %v", err)
+		return Limit{Enabled: false}
+	}
+
+	rlp.conf.RLock()
+	defer rlp.conf.RUnlock()
```

Comment:
I'm not sure how I feel about this. It looks to me like you're letting internal implementation details of `RedisConfStore` leak into `RouteLimitProvider`.

It would be better to encapsulate and separate the two, as was the case previously. `RedisConfStore` should export an API (described by an interface) used by the limit providers. `RedisConfStore` should take care of any needed locking internally.

Reply:
I agree that we should better separate the `LimitProviders` and the `RedisConfStore`. I don't think we can just go back to the way it was previously, since we need 2 different implementations of `GetLimit(req Request) Limit`. But it seems we could put more of this logic in the `RedisConfStore` (including the locking) so that the `LimitProvider` implementations are less aware of how the `RedisConfStore` works.

Reply:
Think I resolved this issue by adding a smaller interface for the `LimitProvider` implementations to depend upon. This way, they don't know how `RedisConfStore` works at all. 

Marking this as resolved, but let me know if you still have concerns regarding this.